lib/matplotlib/image.py

```diff
@@ -521,7 +521,8 @@ def contains(self, mouseevent):
 
     def write_png(self, fname):
         """Write the image to png file with fname"""
-        im = self.to_rgba(self._A, bytes=True, norm=True)
+        im = self.to_rgba(self._A[::-1] if self.origin == 'upper' else self._A,
```

Comment:
@tacaswell ugh, sorry. First I got mixed up on the `origin` polarity, then got my git rebasing all wrong. I think it should look ok now.